src/app/utils/ArrayUtils.ts

```diff
@@ -83,3 +86,48 @@ export class ArrayUtils {
         return ArrayUtils.compareArray(prefix, lst.slice(0, prefix.length));
     }
 }
+
+interface Cell<T> {
+    x: number,
+    y: number,
+    content: T,
+}
+
+export class Table2DWithPossibleNegativeIndices<T extends NonNullable<unknown>> {
+    // This cannot be represented by an array as it may have negative indices
+    // which cannot be iterated over
+    protected content: MGPMap<number, MGPMap<number, T>> = new MGPMap();
```

Comment:
je le dit içi car c'est plus içi, why not just MGPMap<Coord, T>, et Coord ayant le droit d'être négatif, why not CoordMap<T> direct ?

Reply:
car pour dessiner la table on veut itérer sur y d'abord, puis sur x
Tout mettre sous coord ça veut dire ressortir un Coord[] qu'il faut trier, c't'à dire moins efficace que juste itérer sur y puis x